crates/sui-core/src/rpc_index.rs

```diff
@@ -239,10 +239,14 @@ fn balance_delta_merge_operator(
             .expect("Failed to deserialize BalanceIndexInfo from RocksDB - data corruption.");
         result.merge_delta(&delta);
     }
-    Some(
-        bcs::to_bytes(&result)
-            .expect("Failed to deserialize BalanceIndexInfo from RocksDB - data corruption."),
-    )
+    if result.balance_delta == 0 {
+        None
```

Comment:
As per the chat on Slack -- this does not do what you think it does. It signals that the merge failed, not that the merge produced a result that should be compacted.

Here is how to signal deletion, using a separate compaction filter:

https://github.com/MystenLabs/sui/blob/319fe085a2387747dc19d798895d77bd6d594063/crates/sui-indexer-alt-consistent-store/src/schema/balances.rs#L46-L53